### PR TITLE
fix(unmerge): Handle events that don't have a release

### PR DESCRIPTION
**Pull request by bretthoerner**
The test changes are pretty noisy. Constructing an `unmerge` test seems like a lot of work and I wasn't sure we wanted a huge one *just* for this case... so I squeezed it into the existing one (verified it failed without this change). Errr, open to other ideas though.

### CODE REVIEW

**Comment by bretthoerner** [comment]
Sheit forgot to put Fixes SENTRY-6C6

File: src/sentry/tasks/unmerge.py [hunk 1]
```diff
@@ -342,7 +344,7 @@ def repair_tag_data(caches, project, events):
             env_name,
         )
         for key, values in keys.items():
-            tagstore.get_or_create_group_tag_key(
+            _, created = tagstore.get_or_create_group_tag_key(
```
Comment (tkaemming) [inline src/sentry/tasks/unmerge.py]: This assignment isn't necessary/is shadowed below on L358.

**Review by tkaemming — APPROVED** [review]
🍰

File: src/sentry/tasks/unmerge.py (hunk 1 above)
Comment (bretthoerner) [reply src/sentry/tasks/unmerge.py]: Oops ya was part of debugging, will remove.

**Pull request merged**